Tests/Common/SimulatedDevice.cs

```diff
@@ -262,73 +263,86 @@ private bool HandleJoinAccept(LoRaPayloadJoinAccept payload)
+                foreach (var basicsStation in this.simulatedBasicsStations)
                 {
-                    JoinEui = joinRequestPayload.AppEui.ToString("D", null),
-                    msgtype = "jreq",
-                    DevEui = joinRequestPayload.DevEUI.ToString("D", null),
-                    DevNonce = joinRequestPayload.DevNonce.AsUInt16,
-                    MHdr = uint.Parse(joinRequestPayload.MHdr.ToString(), NumberStyles.HexNumber, CultureInfo.InvariantCulture),
-                    MIC = joinRequestPayload.Mic.Value.AsInt32,
-                    DR = joinRequest.RadioMetadata.DataRate,
-                    Freq = joinRequest.RadioMetadata.Frequency.AsUInt64,
-                    upinfo = new
+                    await basicsStation.SerializeAndSendMessageAsync(new
                     {
-                        gpstime = joinRequest.RadioMetadata.UpInfo.GpsTime,
-                        rctx = 10,
-                        rssi = joinRequest.RadioMetadata.UpInfo.ReceivedSignalStrengthIndication,
-                        xtime = joinRequest.RadioMetadata.UpInfo.Xtime,
-                        snr = joinRequest.RadioMetadata.UpInfo.SignalNoiseRatio
-                    }
-                });
-            }
-
-#if DEBUG
-            if (System.Diagnostics.Debugger.IsAttached)
-            {
-                timeout = TimeSpan.FromSeconds(60);
-            }
-#endif
+                        JoinEui = joinRequestPayload.AppEui.ToString("D", null),
+                        msgtype = "jreq",
+                        DevEui = joinRequestPayload.DevEUI.ToString("D", null),
+                        DevNonce = joinRequestPayload.DevNonce.AsUInt16,
+                        MHdr = uint.Parse(joinRequestPayload.MHdr.ToString(), NumberStyles.HexNumber, CultureInfo.InvariantCulture),
+                        MIC = joinRequestPayload.Mic.Value.AsInt32,
+                        DR = joinRequest.RadioMetadata.DataRate,
+                        Freq = joinRequest.RadioMetadata.Frequency.AsUInt64,
+                        upinfo = new
+                        {
+                            gpstime = joinRequest.RadioMetadata.UpInfo.GpsTime,
+                            rctx = 10,
+                            rssi = joinRequest.RadioMetadata.UpInfo.ReceivedSignalStrengthIndication,
+                            xtime = joinRequest.RadioMetadata.UpInfo.Xtime,
+                            snr = joinRequest.RadioMetadata.UpInfo.SignalNoiseRatio
+                        }
+                    });
+                }
 
-            await joinCompleted.WaitAsync(timeout.Value);
+                await joinCompleted.WaitAsync(timeout);
 
-            foreach (var basicsStation in this.simulatedBasicsStations)
-                basicsStation.MessageReceived -= OnMessageReceived;
+                foreach (var basicsStation in this.simulatedBasicsStations)
+                    basicsStation.MessageReceived -= OnMessageReceived;
```

Comment:
If the await fails on [line 339](https://github.com/Azure/iotedge-lorawan-starterkit/pull/1369/files#diff-74cf4610bcf21fec4c6c3b5e5496ea7cc571e6e3ae3ee9adc2280965f3402e7aR339) then these subscriptions won't be revoked. It could have undesired side-effects.

Reply:
See ec604f7434a965776c7882ec4c41694b675146a0 where the subscriptions are revoked irrespective of outcome.